api/jobs/jobs.py

```diff
@@ -83,10 +125,14 @@ def load(cls, e):
         # Don't modify the map
         d = copy.deepcopy(e)
 
-        if d.get('inputs', None):
-            inputs = d['inputs']
-            for x in inputs.keys():
-                inputs[x] = create_filereference_from_dictionary(inputs[x])
+        if d.get('inputs'):
+            input_dict = {}
+
+            for i in d['inputs']:
```

Comment:
Can you elaborate a bit on what this piece of code is doing?
If  two values from `d['inputs']` have the same `input` key, one will override the other. Is it possible/acceptable?